src/app/components/forms/multi-check-box/terra-multi-check-box.component.ts

```diff
@@ -0,0 +1,132 @@
+        }
+    ]
+})
+export class TerraMultiCheckBoxComponent implements OnInit, ControlValueAccessor
+{
+    /**
+     * @description If true, the multi check box will be disabled. Default false.
+     * */
+    @Input() public inputIsDisabled:boolean;
+
+    /**
+     * @description If true, the multi check box will be disabled. Default false.
+     * */
+    @Input() public inputName:string;
+
+    protected valueList:Array<TerraMultiCheckBoxValueInterface> = [];
+
+    @ViewChild('viewChildHeaderCheckbox') protected viewChildHeaderCheckbox:TerraCheckboxComponent;
+    protected headerCheckboxValue:boolean;
+
+    private _isInit:boolean;
+
+    private _langPrefix:string = 'terraMultiCheckBox';
+
+    constructor(public translation:TranslationService)
+    {
+        this.headerCheckboxValue = false;
+    }
+
+    public writeValue(valueList:Array<TerraMultiCheckBoxValueInterface>):void
+    {
+        this.valueList = valueList;
+
+        this.checkHeaderCheckboxState();
+
+        this.onTouchedCallback();
+        this.onChangeCallback(this.valueList);
+    }
+
+    public registerOnChange(fn:any):void
+    {
+        this.onChangeCallback = fn;
+    }
+
+    public registerOnTouched(fn:any):void
+    {
+        this.onTouchedCallback = fn;
+    }
+
+    public ngOnInit():void
+    {
+        if(!this.inputName)
+        {
+            this.inputName = this.translation.translate(this._langPrefix + '.selectAll');
+
+            // this is necessary for language switch
+            this.translation.translationChanged.subscribe(() =>
+            {
+                this.inputName = this.translation.translate(this._langPrefix + '.selectAll');
+            });
+        }
+
+        this._isInit = true;
+    }
+
+    protected onHeaderCheckboxChange(isChecked:boolean):void
+    {
+        this.valueList.forEach((value:TerraMultiCheckBoxValueInterface) =>
+        {
+            value.selected = isChecked;
+        });
+    }
+
+    protected checkHeaderCheckboxState():void
+    {
+        if(!isNullOrUndefined(this.valueList))
+        {
+            let filteredValues:Array<TerraMultiCheckBoxValueInterface> = this.valueList.filter((entry:TerraMultiCheckBoxValueInterface) =>
+            {
+                return entry.selected;
+            });
+
+            this.changeHeaderCheckboxState(filteredValues.length);
+        }
+    }
+
+    private changeHeaderCheckboxState(filteredLength:number):void
+    {
+        if(filteredLength === 0)
+        {
+            this.headerCheckboxValue = false;
+        }
+        else if(filteredLength === this.valueList.length)
+        {
+            this.headerCheckboxValue = true;
+        }
+        else
+        {
+            this.headerCheckboxValue = undefined;
+            this.viewChildHeaderCheckbox.isIndeterminate = true;
+        }
+    }
+
+    private onTouchedCallback:() => void = ():void => undefined;
+
+    private onChangeCallback:(_:any) => void = (_:any):void => undefined;
```

Comment:
Wieso sind die hier unten? Das ist doch eine Variablen declaration?? Und die Initialisierung am besten in den Constructor verschieben.

Reply:
Das sind Funktionen 😄die Variablen stehen oben im Code

Reply:
Nope. Das sind Variablen der Klasse, welche in dem Fall Funktionen bereitstellen.